### PR TITLE
Improvements to CUR and orthogonalizers

### PR DESCRIPTION
**Pull request by rosecers**
Following our work in #86, there were a few edits that needed to be made to improve CUR. I will be closing #86, as we are continuing our work on rank-one updates in a separate repository, but would like to add these small edits to scikit-cosmo.

### CODE REVIEW

File: skcosmo/utils/orthogonalizers.py [hunk 1]
```diff
@@ -25,26 +25,36 @@ def X_orthogonalizer(x1, c=None, x2=None, tol=1e-12):
     :param c: index of the column to orthogonalize by
     :type c: int, less than m
 
-    :param x2: a separate set of columns to orthogonalize by
+    :param x2: a separate set of columns to orthogonalize with column-by-column
```
Comment (Luthaf) [inline skcosmo/utils/orthogonalizers.py:28]: I feel like this is missing the end of the sentence? 

Overall I don't quite understand what this function is doing just by reading the doc. Do you think you could add a bit of explanations?

File: skcosmo/utils/orthogonalizers.py [hunk 2]
```diff
@@ -25,26 +25,36 @@ def X_orthogonalizer(x1, c=None, x2=None, tol=1e-12):
...
     """
 
     if x2 is None and c is not None:
-        x2 = x1[:, [c]]
+        cols = x1[:, [c]]
+    elif x2.shape[0] == x1.shape[0]:
+        cols = np.reshape(x2, (x1.shape[0], -1))
+    else:
+        raise ValueError(
+            f"Orthogonalization only with a matrix containing {x1.shape[0]} entries."
```
Comment (Luthaf) [inline skcosmo/utils/orthogonalizers.py:38]: Could this be extended to say what was expected as well as what the code got?

**Review by Luthaf — COMMENTED** [review]
This looks good overall, I think we could improve a bit the doc/error messages 😃

**Comment by rosecers** [comment]
Sure! I'll add a bit more explanation, didn't know if that was appropriate in the scope of this PR.

**Comment by Luthaf** [comment]
I think that short documentation improvements are always in scope!

File: skcosmo/utils/orthogonalizers.py [hunk 3]
```diff
@@ -14,37 +14,53 @@
...
 
     if x2 is None and c is not None:
-        x2 = x1[:, [c]]
+        cols = x1[:, [c]]
+    elif x2.shape[0] == x1.shape[0]:
+        cols = np.reshape(x2, (x1.shape[0], -1))
+    else:
+        raise ValueError(
+            "You can only orthogonalize a matrix using a vector with the same number of rows."
+            f"Matrix X has {x1.shape[0]} rows., whereas the orthogonalizing matrix has {x2.shape[0]} rows."
```
Comment (Luthaf) [inline skcosmo/utils/orthogonalizers.py:44]:
```suggestion
            "You can only orthogonalize a matrix using a vector with the same number of rows. "
            f"Matrix X has {x1.shape[0]} rows, whereas the orthogonalizing matrix has {x2.shape[0]} rows."
```

**Review by Luthaf — APPROVED** [review]

**Pull request merged**